### DISCUSSION TITLE
Exif copyright value is truncated (vips_image_get_as_string)

### DISCUSSION THREAD

**Discussion by MarcosAndre** [libvips/libvips#3853] [General]
Hello,
I'm using the vips_image_get_as_string to extract copyright from the images, but there are some values that are truncated when the they are too long. I think that the [vips_image_get_as_string](https://github.com/libvips/libvips/blob/67647a6968c263d90f508346cb5c9ac7f3ddb13d/libvips/foreign/exif.c#L293) function only gets the first 256 characters. 

Is it possible to get the entire string using vips? Can you increase the string length?

Thank you in advance.
Best regars,
Marcos André

**Comment by jcupitt** [comment]
Hi @MarcosAndre, the exif library we use might have a limit, though I've not seen it myself. Do you have an example file that has a truncated field?

**Reply by MarcosAndre** [reply]
Yes sure, I will send you an image where the copyright is truncated.
![img-with-lots-of-metadata](https://github.com/libvips/libvips/assets/1064877/05079b07-3ed7-47f2-9001-e1d06cae783b)

**Comment by MarcosAndre** [comment]
[img-with-lots-of-metadata.png.zip](https://github.com/libvips/libvips/files/14308240/img-with-lots-of-metadata.png.zip)

**Comment by jcupitt** [comment]
Ah you're right, there's a 256 byte limit per tag in the libvips exif reader. I'll fix it, thanks!

**Reply by MarcosAndre** [reply]
Great! Thank you! 👍🏼 🥇

**Reply by jcupitt** [reply]
I made an enhancement issue https://github.com/libvips/libvips/issues/3857
